elytron/src/main/java/org/wildfly/extension/elytron/LdapRealmDefinition.java

```diff
@@ -113,7 +113,6 @@
 
         static final SimpleAttributeDefinition RECURSIVE_SEARCH = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.SEARCH_RECURSIVE, ModelType.BOOLEAN, true)
                 .setRequires(ElytronDescriptionConstants.FILTER)
-                .setDefaultValue(ModelNode.TRUE)
```

Comment:
What is the reason for making this change?  By having the default defined which also matches our XML schema it means we can avoid complicating it with the isDefined check.

Reply:
If an attribute A has a default value and requires attribute B then B has to have a default value as well, otherwise it will fail validation. Top-level attributes are being checked for this but I didn't implement it correctly for nested attributes.

If there is an available default value for FILTER and TO I can change that.

Reply:
@darranl how do we proceed with this?